

# A treemap R script produced by the REVIGO server at http://revigo.irb.hr/
# If you found REVIGO useful in your work, please cite the following reference:
# Supek F et al. "REVIGO summarizes and visualizes long lists of Gene Ontology
# terms" PLoS ONE 2011. doi:10.1371/journal.pone.0021800

# author: Anton Kratz <anton.kratz@gmail.com>, RIKEN Omics Science Center, Functional Genomics Technology Team, Japan
# created: Fri, Nov 02, 2012  7:25:52 PM
# last change: Fri, Nov 09, 2012  3:20:01 PM

# -----------------------------------------------------------------------------
# If you don't have the treemap package installed, uncomment the following line:
# install.packages( "treemap" );
library(treemap) 								# treemap package by Martijn Tennekes

# Set the working directory if necessary
# setwd("C:/Users/username/workingdir");

# --------------------------------------------------------------------------
# Here is your data from REVIGO. Scroll down for plot configuration options.

revigo.names <- c("term_ID","description","freqInDbPercent","abslog10pvalue","uniqueness","dispensability","representative");
revigo.data <- rbind(c("GO:0019752","carboxylic acid metabolic process",2.886,7.6198,0.646,0.000,"carboxylic acid metabolism"),
c("GO:0072524","pyridine-containing compound metabolic process",0.204,2.7670,0.829,0.103,"carboxylic acid metabolism"),
c("GO:0044710","single-organism metabolic process",26.477,3.6021,0.898,0.112,"carboxylic acid metabolism"),
c("GO:0051186","cofactor metabolic process",0.798,4.1739,0.892,0.119,"carboxylic acid metabolism"),
c("GO:0005975","carbohydrate metabolic process",3.046,1.5619,0.887,0.138,"carboxylic acid metabolism"),
c("GO:0030638","polyketide metabolic process",0.009,1.5275,0.861,0.152,"carboxylic acid metabolism"),
c("GO:0046149","pigment catabolic process",0.010,3.2147,0.812,0.153,"carboxylic acid metabolism"),
c("GO:0006091","generation of precursor metabolites and energy",6.571,8.2757,0.868,0.160,"carboxylic acid metabolism"),
c("GO:0046184","aldehyde biosynthetic process",0.021,2.0367,0.774,0.162,"carboxylic acid metabolism"),
c("GO:0032800","receptor biosynthetic process",0.030,1.5387,0.869,0.164,"carboxylic acid metabolism"),
c("GO:0043603","cellular amide metabolic process",0.480,1.4170,0.861,0.178,"carboxylic acid metabolism"),
c("GO:0006081","cellular aldehyde metabolic process",0.106,1.6082,0.827,0.188,"carboxylic acid metabolism"),
c("GO:0042440","pigment metabolic process",0.165,1.5488,0.844,0.196,"carboxylic acid metabolism"),
c("GO:0005996","monosaccharide metabolic process",0.738,4.5686,0.762,0.231,"carboxylic acid metabolism"),
c("GO:0044237","cellular metabolic process",42.121,1.5058,0.886,0.249,"carboxylic acid metabolism"),
c("GO:1901564","organonitrogen compound metabolic process",11.844,1.7807,0.827,0.261,"carboxylic acid metabolism"),
c("GO:0046950","cellular ketone body metabolic process",0.016,1.5275,0.802,0.281,"carboxylic acid metabolism"),
c("GO:0015949","nucleobase-containing small molecule interconversion",0.021,1.5387,0.780,0.287,"carboxylic acid metabolism"),
c("GO:0016139","glycoside catabolic process",0.008,1.6619,0.792,0.293,"carboxylic acid metabolism"),
c("GO:0022904","respiratory electron transport chain",4.683,7.3010,0.729,0.296,"carboxylic acid metabolism"),
c("GO:0033013","tetrapyrrole metabolic process",0.173,2.2534,0.831,0.300,"carboxylic acid metabolism"),
c("GO:0006629","lipid metabolic process",3.159,3.4559,0.792,0.300,"carboxylic acid metabolism"),
c("GO:0070221","sulfide oxidation, using sulfide:quinone oxidoreductase",0.006,1.8265,0.832,0.307,"carboxylic acid metabolism"),
c("GO:0044711","single-organism biosynthetic process",4.776,2.0540,0.761,0.320,"carboxylic acid metabolism"),
c("GO:0046951","ketone body biosynthetic process",0.009,1.6619,0.757,0.326,"carboxylic acid metabolism"),
c("GO:0006739","NADP metabolic process",0.087,3.7447,0.697,0.326,"carboxylic acid metabolism"),
c("GO:0006766","vitamin metabolic process",0.209,1.6280,0.786,0.356,"carboxylic acid metabolism"),
c("GO:0055114","oxidation-reduction process",10.933,11.2757,0.770,0.372,"carboxylic acid metabolism"),
c("GO:0010565","regulation of cellular ketone metabolic process",0.322,1.6216,0.700,0.373,"carboxylic acid metabolism"),
c("GO:0072522","purine-containing compound biosynthetic process",1.805,1.7387,0.738,0.381,"carboxylic acid metabolism"),
c("GO:0044282","small molecule catabolic process",0.575,4.5086,0.718,0.397,"carboxylic acid metabolism"),
c("GO:0090407","organophosphate biosynthetic process",2.638,1.6676,0.772,0.403,"carboxylic acid metabolism"),
c("GO:0050994","regulation of lipid catabolic process",0.089,3.3188,0.706,0.448,"carboxylic acid metabolism"),
c("GO:0090410","malonate catabolic process",0.001,1.3951,0.751,0.452,"carboxylic acid metabolism"),
c("GO:0044281","small molecule metabolic process",13.539,7.0969,0.765,0.464,"carboxylic acid metabolism"),
c("GO:0072363","regulation of glycolytic by positive regulation of transcription from RNA polymerase II promoter",0.001,1.3951,0.681,0.472,"carboxylic acid metabolism"),
c("GO:0030209","dermatan sulfate catabolic process",0.001,1.3951,0.709,0.473,"carboxylic acid metabolism"),
c("GO:0006082","organic acid metabolic process",3.105,6.2076,0.702,0.493,"carboxylic acid metabolism"),
c("GO:0019418","sulfide oxidation",0.006,1.8265,0.832,0.498,"carboxylic acid metabolism"),
c("GO:0019402","galactitol metabolic process",0.005,1.3951,0.772,0.516,"carboxylic acid metabolism"),
c("GO:0006438","valyl-tRNA aminoacylation",0.009,2.7905,0.733,0.525,"carboxylic acid metabolism"),
c("GO:0009082","branched-chain amino acid biosynthetic process",0.009,1.3951,0.699,0.525,"carboxylic acid metabolism"),
c("GO:0000379","tRNA-type intron splice site recognition and cleavage",0.005,1.3951,0.882,0.532,"carboxylic acid metabolism"),
c("GO:0019322","pentose biosynthetic process",0.008,3.6021,0.756,0.536,"carboxylic acid metabolism"),
c("GO:0016042","lipid catabolic process",0.537,2.8962,0.715,0.546,"carboxylic acid metabolism"),
c("GO:0019388","galactose catabolic process",0.011,1.8265,0.768,0.550,"carboxylic acid metabolism"),
c("GO:0006099","tricarboxylic acid cycle",0.104,4.1427,0.804,0.550,"carboxylic acid metabolism"),
c("GO:0034440","lipid oxidation",0.158,1.5137,0.736,0.573,"carboxylic acid metabolism"),
c("GO:0052695","cellular glucuronidation",0.021,1.5275,0.696,0.574,"carboxylic acid metabolism"),
c("GO:0044712","single-organism catabolic process",5.557,2.5003,0.726,0.583,"carboxylic acid metabolism"),
c("GO:0019321","pentose metabolic process",0.028,3.3468,0.801,0.589,"carboxylic acid metabolism"),
c("GO:0006063","uronic acid metabolic process",0.025,1.4141,0.698,0.592,"carboxylic acid metabolism"),
c("GO:0009229","thiamine diphosphate biosynthetic process",0.008,1.3951,0.754,0.610,"carboxylic acid metabolism"),
c("GO:0006665","sphingolipid metabolic process",0.312,1.4522,0.701,0.612,"carboxylic acid metabolism"),
c("GO:0009123","nucleoside monophosphate metabolic process",5.669,1.4299,0.646,0.625,"carboxylic acid metabolism"),
c("GO:0016115","terpenoid catabolic process",0.007,2.3261,0.745,0.626,"carboxylic acid metabolism"),
c("GO:0006669","sphinganine-1-phosphate biosynthetic process",0.001,1.3951,0.730,0.628,"carboxylic acid metabolism"),
c("GO:0006787","porphyrin-containing compound catabolic process",0.027,3.2147,0.754,0.628,"carboxylic acid metabolism"),
c("GO:0072366","regulation of cellular ketone metabolic process by positive regulation of transcription from RNA polymerase II promoter",0.001,1.3951,0.763,0.630,"carboxylic acid metabolism"),
c("GO:0008300","isoprenoid catabolic process",0.011,1.8265,0.741,0.648,"carboxylic acid metabolism"),
c("GO:0042357","thiamine diphosphate metabolic process",0.010,1.3951,0.792,0.649,"carboxylic acid metabolism"),
c("GO:0006636","unsaturated fatty acid biosynthetic process",0.108,1.7610,0.628,0.651,"carboxylic acid metabolism"),
c("GO:0051187","cofactor catabolic process",0.047,1.9642,0.803,0.651,"carboxylic acid metabolism"),
c("GO:0016052","carbohydrate catabolic process",0.480,3.5850,0.793,0.661,"carboxylic acid metabolism"),
c("GO:0006707","cholesterol catabolic process",0.019,1.6619,0.706,0.668,"carboxylic acid metabolism"),
c("GO:0006749","glutathione metabolic process",0.149,2.9706,0.683,0.671,"carboxylic acid metabolism"),
c("GO:0046456","icosanoid biosynthetic process",0.098,1.4622,0.675,0.696,"carboxylic acid metabolism"),
c("GO:0050996","positive regulation of lipid catabolic process",0.036,2.8601,0.712,0.696,"carboxylic acid metabolism"),
c("GO:0043648","dicarboxylic acid metabolic process",0.213,1.9626,0.712,0.696,"carboxylic acid metabolism"),
c("GO:0043279","response to alkaloid",0.450,2.8182,0.951,0.000,"response to alkaloid"),
c("GO:0070980","biphenyl catabolic process",0.006,1.3951,0.816,0.294,"response to alkaloid"),
c("GO:0071449","cellular response to lipid hydroperoxide",0.001,1.3951,0.928,0.313,"response to alkaloid"),
c("GO:0071276","cellular response to cadmium ion",0.032,1.5275,0.923,0.332,"response to alkaloid"),
c("GO:0071395","cellular response to jasmonic acid stimulus",0.005,2.0367,0.925,0.412,"response to alkaloid"),
c("GO:0071361","cellular response to ethanol",0.022,1.6619,0.924,0.422,"response to alkaloid"),
c("GO:0009753","response to jasmonic acid",0.005,2.0367,0.960,0.491,"response to alkaloid"),
c("GO:0071384","cellular response to corticosteroid stimulus",0.112,1.4722,0.913,0.520,"response to alkaloid"),
c("GO:0070541","response to platinum ion",0.006,1.3951,0.965,0.564,"response to alkaloid"),
c("GO:0042631","cellular response to water deprivation",0.006,1.3951,0.883,0.566,"response to alkaloid"),
c("GO:0035094","response to nicotine",0.207,2.7212,0.949,0.667,"response to alkaloid"),
c("GO:0006808","regulation of nitrogen utilization",0.009,1.3951,0.961,0.015,"regulation of nitrogen utilization"),
c("GO:0019740","nitrogen utilization",0.009,1.3951,0.977,0.015,"nitrogen utilization"),
c("GO:0006848","pyruvate transport",0.010,4.1871,0.889,0.015,"pyruvate transport"),
c("GO:0060454","positive regulation of gastric acid secretion",0.003,1.3951,0.859,0.178,"pyruvate transport"),
c("GO:0014806","smooth muscle hyperplasia",0.007,1.3951,0.934,0.305,"pyruvate transport"),
c("GO:1902600","hydrogen ion transmembrane transport",3.156,4.1675,0.797,0.372,"pyruvate transport"),
c("GO:0006839","mitochondrial transport",0.326,2.1124,0.882,0.376,"pyruvate transport"),
c("GO:0047497","mitochondrion transport along microtubule",0.018,1.6619,0.846,0.501,"pyruvate transport"),
c("GO:0006818","hydrogen transport",3.252,3.2924,0.881,0.502,"pyruvate transport"),
c("GO:0042776","mitochondrial ATP synthesis coupled proton transport",0.018,3.8239,0.634,0.550,"pyruvate transport"),
c("GO:0046390","ribose phosphate biosynthetic process",1.781,2.1379,0.726,0.560,"pyruvate transport"),
c("GO:0055085","transmembrane transport",7.528,1.8948,0.801,0.575,"pyruvate transport"),
c("GO:2000300","regulation of synaptic vesicle exocytosis",0.049,1.5275,0.787,0.624,"pyruvate transport"),
c("GO:0006820","anion transport",1.726,1.6317,0.874,0.626,"pyruvate transport"),
c("GO:1902445","regulation of mitochondrial membrane permeability involved in programmed necrotic cell death",0.008,1.3951,0.809,0.633,"pyruvate transport"),
c("GO:0015727","lactate transport",0.008,2.0367,0.891,0.633,"pyruvate transport"),
c("GO:0035873","lactate transmembrane transport",0.008,2.0367,0.850,0.633,"pyruvate transport"),
c("GO:0006850","mitochondrial pyruvate transport",0.009,2.7905,0.872,0.638,"pyruvate transport"),
c("GO:0015718","monocarboxylic acid transport",0.270,2.0565,0.867,0.658,"pyruvate transport"),
c("GO:1902476","chloride transmembrane transport",0.222,1.8383,0.823,0.664,"pyruvate transport"),
c("GO:0006811","ion transport",7.803,1.6720,0.870,0.673,"pyruvate transport"),
c("GO:0015911","plasma membrane long-chain fatty acid transport",0.015,1.5275,0.834,0.674,"pyruvate transport"),
c("GO:1902582","single-organism intracellular transport",2.514,1.9027,0.863,0.685,"pyruvate transport"),
c("GO:0032981","mitochondrial respiratory chain complex I assembly",0.018,1.9642,0.924,0.034,"mitochondrial respiratory chain complex I assembly"),
c("GO:0032534","regulation of microvillus assembly",0.007,1.3951,0.890,0.324,"mitochondrial respiratory chain complex I assembly"),
c("GO:0097031","mitochondrial respiratory chain complex I biogenesis",0.018,1.9642,0.968,0.413,"mitochondrial respiratory chain complex I assembly"),
c("GO:0036034","mediator complex assembly",0.001,1.3951,0.968,0.450,"mitochondrial respiratory chain complex I assembly"),
c("GO:2001178","positive regulation of mediator complex assembly",0.001,1.3951,0.941,0.450,"mitochondrial respiratory chain complex I assembly"),
c("GO:0010257","NADH dehydrogenase complex assembly",0.018,1.9642,0.965,0.523,"mitochondrial respiratory chain complex I assembly"),
c("GO:0043653","mitochondrial fragmentation involved in apoptotic process",0.011,1.5275,0.893,0.566,"mitochondrial respiratory chain complex I assembly"),
c("GO:0032976","release of matrix enzymes from mitochondria",0.001,1.3951,0.906,0.593,"mitochondrial respiratory chain complex I assembly"),
c("GO:0000266","mitochondrial fission",0.042,1.6874,0.927,0.611,"mitochondrial respiratory chain complex I assembly"),
c("GO:0033108","mitochondrial respiratory chain complex assembly",0.040,1.4722,0.921,0.642,"mitochondrial respiratory chain complex I assembly"),
c("GO:0050999","regulation of nitric-oxide synthase activity",0.060,3.9208,0.894,0.050,"regulation of nitric-oxide synthase activity"),
c("GO:0060263","regulation of respiratory burst",0.015,1.5275,0.840,0.175,"regulation of nitric-oxide synthase activity"),
c("GO:0035229","positive regulation of glutamate-cysteine ligase activity",0.001,1.3951,0.917,0.297,"regulation of nitric-oxide synthase activity"),
c("GO:0043095","regulation of GTP cyclohydrolase I activity",0.002,1.3951,0.915,0.308,"regulation of nitric-oxide synthase activity"),
c("GO:0035563","positive regulation of chromatin binding",0.008,1.3951,0.961,0.309,"regulation of nitric-oxide synthase activity"),
c("GO:0051341","regulation of oxidoreductase activity",0.137,2.4389,0.894,0.398,"regulation of nitric-oxide synthase activity"),
c("GO:0035227","regulation of glutamate-cysteine ligase activity",0.001,1.3951,0.917,0.581,"regulation of nitric-oxide synthase activity"),
c("GO:0010817","regulation of hormone levels",1.029,2.5768,0.930,0.054,"regulation of hormone levels"),
c("GO:0006702","androgen biosynthetic process",0.026,2.3429,0.684,0.378,"regulation of hormone levels"),
c("GO:0032287","peripheral nervous system myelin maintenance",0.048,1.5275,0.808,0.396,"regulation of hormone levels"),
c("GO:0036149","phosphatidylinositol acyl-chain remodeling",0.018,1.4141,0.754,0.397,"regulation of hormone levels"),
c("GO:0003051","angiotensin-mediated drinking behavior",0.007,1.3951,0.898,0.403,"regulation of hormone levels"),
c("GO:0010883","regulation of lipid storage",0.088,1.7174,0.870,0.417,"regulation of hormone levels"),
c("GO:0007588","excretion",0.129,1.4622,0.932,0.423,"regulation of hormone levels"),
c("GO:0010634","positive regulation of epithelial cell migration",0.119,1.5137,0.792,0.436,"regulation of hormone levels"),
c("GO:0015920","lipopolysaccharide transport",0.002,1.3951,0.911,0.539,"regulation of hormone levels"),
c("GO:0006614","SRP-dependent cotranslational protein targeting to membrane",0.150,1.4289,0.853,0.597,"regulation of hormone levels"),
c("GO:0045759","negative regulation of action potential",0.008,1.3951,0.927,0.601,"regulation of hormone levels"),
c("GO:1900053","negative regulation of retinoic acid biosynthetic process",0.001,1.3951,0.640,0.633,"regulation of hormone levels"),
c("GO:0008206","bile acid metabolic process",0.077,1.9038,0.675,0.657,"regulation of hormone levels"),
c("GO:0042445","hormone metabolic process",0.448,2.0195,0.875,0.663,"regulation of hormone levels"),
c("GO:0070972","protein localization to endoplasmic reticulum",0.196,1.5032,0.912,0.677,"regulation of hormone levels"),
c("GO:0033004","negative regulation of mast cell activation",0.024,3.6021,0.835,0.073,"negative regulation of mast cell activation"),
c("GO:0048385","regulation of retinoic acid receptor signaling pathway",0.073,2.2013,0.870,0.113,"negative regulation of mast cell activation"),
c("GO:0002504","antigen processing and presentation of peptide or polysaccharide antigen via MHC class II",4.627,2.1649,0.966,0.152,"negative regulation of mast cell activation"),
c("GO:0050690","regulation of defense response to virus by virus",0.033,1.9731,0.921,0.156,"negative regulation of mast cell activation"),
c("GO:0002448","mast cell mediated immunity",0.081,1.9642,0.957,0.170,"negative regulation of mast cell activation"),
c("GO:0007168","receptor guanylyl cyclase signaling pathway",0.013,2.0367,0.891,0.189,"negative regulation of mast cell activation"),
c("GO:0019886","antigen processing and presentation of exogenous peptide antigen via MHC class II",0.130,2.3206,0.966,0.216,"negative regulation of mast cell activation"),
c("GO:0019884","antigen processing and presentation of exogenous antigen",0.303,1.4943,0.971,0.237,"negative regulation of mast cell activation"),
c("GO:0007268","synaptic transmission",1.611,1.7044,0.875,0.255,"negative regulation of mast cell activation"),
c("GO:0035235","ionotropic glutamate receptor signaling pathway",0.106,1.6619,0.875,0.278,"negative regulation of mast cell activation"),
c("GO:2000474","regulation of opioid receptor signaling pathway",0.006,1.3951,0.890,0.314,"negative regulation of mast cell activation"),
c("GO:0009589","detection of UV",0.005,1.3951,0.977,0.330,"negative regulation of mast cell activation"),
c("GO:0044406","adhesion of symbiont to host",0.007,1.8265,0.986,0.338,"negative regulation of mast cell activation"),
c("GO:0002432","granuloma formation",0.002,1.3951,0.957,0.360,"negative regulation of mast cell activation"),
c("GO:0050857","positive regulation of antigen receptor-mediated signaling pathway",0.052,1.8265,0.850,0.361,"negative regulation of mast cell activation"),
c("GO:0007267","cell-cell signaling",2.522,1.4603,0.878,0.371,"negative regulation of mast cell activation"),
c("GO:0043046","DNA methylation involved in gamete generation",0.036,1.4141,0.802,0.374,"negative regulation of mast cell activation"),
c("GO:0046745","viral capsid secondary envelopment",0.001,1.3951,0.958,0.425,"negative regulation of mast cell activation"),
c("GO:0051037","regulation of transcription during meiosis",0.013,1.3951,0.791,0.429,"negative regulation of mast cell activation"),
c("GO:0002495","antigen processing and presentation of peptide antigen via MHC class II",0.134,2.2027,0.970,0.442,"negative regulation of mast cell activation"),
c("GO:0002358","B cell homeostatic proliferation",0.010,1.3951,0.895,0.465,"negative regulation of mast cell activation"),
c("GO:0051038","negative regulation of transcription during meiosis",0.001,1.3951,0.801,0.526,"negative regulation of mast cell activation"),
c("GO:0060340","positive regulation of type I interferon-mediated signaling pathway",0.021,1.5275,0.834,0.538,"negative regulation of mast cell activation"),
c("GO:0035669","TRAM-dependent toll-like receptor 4 signaling pathway",0.006,1.3951,0.867,0.563,"negative regulation of mast cell activation"),
c("GO:0001777","T cell homeostatic proliferation",0.014,1.3951,0.860,0.576,"negative regulation of mast cell activation"),
c("GO:1990117","B cell receptor apoptotic signaling pathway",0.001,1.3951,0.878,0.613,"negative regulation of mast cell activation"),
c("GO:0002719","negative regulation of cytokine production involved in immune response",0.023,1.6619,0.863,0.621,"negative regulation of mast cell activation"),
c("GO:0002373","plasmacytoid dendritic cell cytokine production",0.001,1.3951,0.914,0.648,"negative regulation of mast cell activation"),
c("GO:0051156","glucose 6-phosphate metabolic process",0.024,2.9281,0.851,0.084,"glucose 6-phosphate metabolism"),
c("GO:0016137","glycoside metabolic process",0.030,2.6055,0.834,0.244,"glucose 6-phosphate metabolism"),
c("GO:0016925","protein sumoylation",0.064,1.6082,0.893,0.092,"protein sumoylation"),
c("GO:0010731","protein glutathionylation",0.001,1.3951,0.915,0.224,"protein sumoylation"),
c("GO:0048539","bone marrow development",0.008,2.0367,0.914,0.097,"bone marrow development"),
c("GO:1902337","regulation of apoptotic process involved in morphogenesis",0.035,1.6619,0.857,0.220,"bone marrow development"),
c("GO:0021762","substantia nigra development",0.055,1.7610,0.919,0.285,"bone marrow development"),
c("GO:0001949","sebaceous gland cell differentiation",0.001,1.3951,0.894,0.292,"bone marrow development"),
c("GO:0043587","tongue morphogenesis",0.027,1.5275,0.923,0.327,"bone marrow development"),
c("GO:0072229","metanephric proximal convoluted tubule development",0.006,1.3951,0.923,0.335,"bone marrow development"),
c("GO:1902512","positive regulation of apoptotic DNA fragmentation",0.001,1.3951,0.783,0.347,"bone marrow development"),
c("GO:2001037","positive regulation of tongue muscle cell differentiation",0.003,1.3951,0.850,0.364,"bone marrow development"),
c("GO:1990001","inhibition of cysteine-type endopeptidase activity involved in apoptotic process",0.003,1.3951,0.833,0.368,"bone marrow development"),
c("GO:0070407","oxidation-dependent protein catabolic process",0.008,1.3951,0.853,0.380,"bone marrow development"),
c("GO:0060465","pharynx development",0.002,1.3951,0.935,0.396,"bone marrow development"),
c("GO:0010734","negative regulation of protein glutathionylation",0.001,1.3951,0.863,0.398,"bone marrow development"),
c("GO:1901208","negative regulation of heart looping",0.001,1.3951,0.890,0.409,"bone marrow development"),
c("GO:0035470","positive regulation of vascular wound healing",0.003,1.3951,0.888,0.433,"bone marrow development"),
c("GO:0045358","negative regulation of interferon-beta biosynthetic process",0.009,1.3951,0.787,0.440,"bone marrow development"),
c("GO:0007521","muscle cell fate determination",0.007,1.3951,0.899,0.463,"bone marrow development"),
c("GO:0043105","negative regulation of GTP cyclohydrolase I activity",0.002,1.3951,0.913,0.465,"bone marrow development"),
c("GO:0061034","olfactory bulb mitral cell layer development",0.003,1.3951,0.926,0.499,"bone marrow development"),
c("GO:0048343","paraxial mesodermal cell fate commitment",0.003,1.3951,0.881,0.508,"bone marrow development"),
c("GO:0097341","zymogen inhibition",0.003,1.3951,0.872,0.638,"bone marrow development"),
c("GO:0098528","skeletal muscle fiber differentiation",0.003,1.3951,0.882,0.640,"bone marrow development"),
c("GO:1901207","regulation of heart looping",0.001,1.3951,0.904,0.643,"bone marrow development"),
c("GO:1901211","negative regulation of cardiac chamber formation",0.001,1.3951,0.893,0.643,"bone marrow development"),
c("GO:0021536","diencephalon development",0.350,1.5110,0.911,0.674,"bone marrow development"));

stuff <- data.frame(revigo.data);
names(stuff) <- revigo.names;

stuff$abslog10pvalue <- as.numeric( as.character(stuff$abslog10pvalue) );
stuff$freqInDbPercent <- as.numeric( as.character(stuff$freqInDbPercent) );
stuff$uniqueness <- as.numeric( as.character(stuff$uniqueness) );
stuff$dispensability <- as.numeric( as.character(stuff$dispensability) );

# by default, outputs to a PDF file
pdf( file="revigo_treemap.pdf", width=16, height=9 ) # width and height are in inches

# check the tmPlot command documentation for all possible parameters - there are a lot more
tmPlot(
	stuff,
	index = c("representative","description"),
	vSize = "abslog10pvalue",
	type = "categorical",
	vColor = "representative",
	title = "REVIGO Gene Ontology treemap",
	inflate.labels = FALSE,      # set this to TRUE for space-filling group labels - good for posters
	lowerbound.cex.labels = 0,   # try to draw as many labels as possible (still, some small squares may not get a label)
	bg.labels = "#CCCCCCAA",     # define background color of group labels
												       # "#CCCCCC00" is fully transparent, "#CCCCCCAA" is semi-transparent grey, NA is opaque
	position.legend = "none"
)

dev.off()
