

# A treemap R script produced by the REVIGO server at http://revigo.irb.hr/
# If you found REVIGO useful in your work, please cite the following reference:
# Supek F et al. "REVIGO summarizes and visualizes long lists of Gene Ontology
# terms" PLoS ONE 2011. doi:10.1371/journal.pone.0021800

# author: Anton Kratz <anton.kratz@gmail.com>, RIKEN Omics Science Center, Functional Genomics Technology Team, Japan
# created: Fri, Nov 02, 2012  7:25:52 PM
# last change: Fri, Nov 09, 2012  3:20:01 PM

# -----------------------------------------------------------------------------
# If you don't have the treemap package installed, uncomment the following line:
# install.packages( "treemap" );
library(treemap) 								# treemap package by Martijn Tennekes

# Set the working directory if necessary
# setwd("C:/Users/username/workingdir");

# --------------------------------------------------------------------------
# Here is your data from REVIGO. Scroll down for plot configuration options.

revigo.names <- c("term_ID","description","freqInDbPercent","abslog10pvalue","uniqueness","dispensability","representative");
revigo.data <- rbind(c("GO:0010092","specification of organ identity",0.103,4.0000,0.613,0.000,"specification of organ identity"),
c("GO:0050975","sensory perception of touch",0.006,1.8952,0.800,0.193,"specification of organ identity"),
c("GO:0034381","plasma lipoprotein particle clearance",0.076,1.7144,0.755,0.228,"specification of organ identity"),
c("GO:0032735","positive regulation of interleukin-12 production",0.077,2.0106,0.714,0.229,"specification of organ identity"),
c("GO:2000744","positive regulation of anterior head development",0.006,1.8952,0.751,0.229,"specification of organ identity"),
c("GO:0007585","respiratory gaseous exchange",0.131,2.0092,0.768,0.238,"specification of organ identity"),
c("GO:0048880","sensory system development",0.005,1.4235,0.749,0.248,"specification of organ identity"),
c("GO:0042636","negative regulation of hair cycle",0.001,1.8952,0.768,0.295,"specification of organ identity"),
c("GO:0045165","cell fate commitment",0.620,2.2218,0.685,0.297,"specification of organ identity"),
c("GO:0048585","negative regulation of response to stimulus",2.501,1.5771,0.873,0.298,"specification of organ identity"),
c("GO:0060027","convergent extension involved in gastrulation",0.003,1.4235,0.709,0.308,"specification of organ identity"),
c("GO:0007569","cell aging",0.203,1.3898,0.725,0.315,"specification of organ identity"),
c("GO:0021544","subpallium development",0.047,3.0000,0.646,0.332,"specification of organ identity"),
c("GO:0003016","respiratory system process",0.057,1.6670,0.774,0.351,"specification of organ identity"),
c("GO:0045061","thymic T cell selection",0.062,2.0862,0.646,0.372,"specification of organ identity"),
c("GO:0014824","artery smooth muscle contraction",0.018,1.3014,0.767,0.377,"specification of organ identity"),
c("GO:0007614","short-term memory",0.027,1.5969,0.769,0.386,"specification of organ identity"),
c("GO:0030324","lung development",0.575,3.0410,0.607,0.401,"specification of organ identity"),
c("GO:0007389","pattern specification process",1.143,1.5865,0.658,0.403,"specification of organ identity"),
c("GO:0048663","neuron fate commitment",0.153,2.4750,0.621,0.404,"specification of organ identity"),
c("GO:0060541","respiratory system development",0.648,2.6925,0.656,0.410,"specification of organ identity"),
c("GO:0045163","clustering of voltage-gated potassium channels",0.003,1.8952,0.684,0.413,"specification of organ identity"),
c("GO:0021501","prechordal plate formation",0.003,1.8952,0.705,0.415,"specification of organ identity"),
c("GO:0060384","innervation",0.071,2.0862,0.672,0.416,"specification of organ identity"),
c("GO:2001240","negative regulation of extrinsic apoptotic signaling pathway in absence of ligand",0.093,1.4665,0.793,0.421,"specification of organ identity"),
c("GO:0033564","anterior/posterior axon guidance",0.008,1.8952,0.612,0.435,"specification of organ identity"),
c("GO:0060040","retinal bipolar neuron differentiation",0.006,1.8952,0.643,0.445,"specification of organ identity"),
c("GO:0048936","peripheral nervous system neuron axonogenesis",0.003,1.5969,0.667,0.449,"specification of organ identity"),
c("GO:0035295","tube development",1.620,1.5471,0.647,0.452,"specification of organ identity"),
c("GO:0072564","blood microparticle formation",0.016,1.8952,0.707,0.456,"specification of organ identity"),
c("GO:2000334","positive regulation of blood microparticle formation",0.016,1.8952,0.654,0.456,"specification of organ identity"),
c("GO:0021675","nerve development",0.234,1.9108,0.653,0.456,"specification of organ identity"),
c("GO:0048514","blood vessel morphogenesis",1.146,2.4881,0.595,0.466,"specification of organ identity"),
c("GO:0060592","mammary gland formation",0.025,2.6326,0.662,0.470,"specification of organ identity"),
c("GO:0060529","squamous basal epithelial stem cell differentiation involved in prostate gland acinus development",0.006,1.5969,0.669,0.471,"specification of organ identity"),
c("GO:0043588","skin development",0.701,2.5935,0.630,0.477,"specification of organ identity"),
c("GO:0008544","epidermis development",0.590,2.3251,0.612,0.478,"specification of organ identity"),
c("GO:0036301","macrophage colony-stimulating factor production",0.002,1.8952,0.795,0.484,"specification of organ identity"),
c("GO:0071611","granulocyte colony-stimulating factor production",0.002,1.8952,0.795,0.484,"specification of organ identity"),
c("GO:2000346","negative regulation of hepatocyte proliferation",0.009,1.8952,0.582,0.484,"specification of organ identity"),
c("GO:0030031","cell projection assembly",0.692,1.4278,0.888,0.485,"specification of organ identity"),
c("GO:0072359","circulatory system development",2.260,2.2140,0.621,0.506,"specification of organ identity"),
c("GO:0007601","visual perception",0.288,1.6899,0.747,0.510,"specification of organ identity"),
c("GO:0060875","lateral semicircular canal development",0.002,1.8952,0.717,0.512,"specification of organ identity"),
c("GO:0032808","lacrimal gland development",0.024,1.5969,0.687,0.512,"specification of organ identity"),
c("GO:0021517","ventral spinal cord development",0.108,2.6459,0.629,0.512,"specification of organ identity"),
c("GO:0032604","granulocyte macrophage colony-stimulating factor production",0.013,1.3014,0.776,0.535,"specification of organ identity"),
c("GO:0045665","negative regulation of neuron differentiation",0.168,1.3098,0.553,0.538,"specification of organ identity"),
c("GO:0021860","pyramidal neuron development",0.017,1.3014,0.635,0.543,"specification of organ identity"),
c("GO:0071109","superior temporal gyrus development",0.002,1.5969,0.699,0.547,"specification of organ identity"),
c("GO:0009888","tissue development",4.199,1.4268,0.672,0.549,"specification of organ identity"),
c("GO:0035176","social behavior",0.109,1.5800,0.950,0.549,"specification of organ identity"),
c("GO:0032645","regulation of granulocyte macrophage colony-stimulating factor production",0.013,1.3014,0.751,0.552,"specification of organ identity"),
c("GO:0048645","organ formation",0.168,2.9666,0.620,0.558,"specification of organ identity"),
c("GO:0060602","branch elongation of an epithelium",0.062,1.8781,0.649,0.565,"specification of organ identity"),
c("GO:0021510","spinal cord development",0.198,1.8502,0.624,0.569,"specification of organ identity"),
c("GO:0071657","positive regulation of granulocyte colony-stimulating factor production",0.002,1.8952,0.758,0.573,"specification of organ identity"),
c("GO:0060896","neural plate pattern specification",0.021,2.3686,0.723,0.575,"specification of organ identity"),
c("GO:0007417","central nervous system development",2.238,1.5746,0.589,0.580,"specification of organ identity"),
c("GO:0030890","positive regulation of B cell proliferation",0.075,1.7652,0.825,0.586,"specification of organ identity"),
c("GO:0021999","neural plate anterior/posterior regionalization",0.014,2.8041,0.725,0.589,"specification of organ identity"),
c("GO:0030901","midbrain development",0.190,1.5800,0.622,0.595,"specification of organ identity"),
c("GO:0048704","embryonic skeletal system morphogenesis",0.237,2.3152,0.599,0.596,"specification of organ identity"),
c("GO:0045892","negative regulation of transcription, DNA-templated",2.182,1.5322,0.792,0.597,"specification of organ identity"),
c("GO:0021798","forebrain dorsal/ventral pattern formation",0.017,1.5969,0.655,0.597,"specification of organ identity"),
c("GO:0010573","vascular endothelial growth factor production",0.071,1.7652,0.752,0.599,"specification of organ identity"),
c("GO:2001223","negative regulation of neuron migration",0.050,1.3014,0.563,0.603,"specification of organ identity"),
c("GO:0032730","positive regulation of interleukin-1 alpha production",0.006,1.5969,0.751,0.604,"specification of organ identity"),
c("GO:0097065","anterior head development",0.006,1.8952,0.784,0.617,"specification of organ identity"),
c("GO:0060913","cardiac cell fate determination",0.006,1.8952,0.680,0.621,"specification of organ identity"),
c("GO:0048934","peripheral nervous system neuron differentiation",0.019,1.3014,0.670,0.625,"specification of organ identity"),
c("GO:0021859","pyramidal neuron differentiation",0.018,1.3014,0.656,0.629,"specification of organ identity"),
c("GO:0032615","interleukin-12 production",0.140,1.3996,0.740,0.629,"specification of organ identity"),
c("GO:0001706","endoderm formation",0.092,1.6670,0.655,0.635,"specification of organ identity"),
c("GO:0021953","central nervous system neuron differentiation",0.427,1.6388,0.586,0.636,"specification of organ identity"),
c("GO:0050910","detection of mechanical stimulus involved in sensory perception of sound",0.027,1.4235,0.746,0.639,"specification of organ identity"),
c("GO:0021756","striatum development",0.032,2.0106,0.651,0.639,"specification of organ identity"),
c("GO:0021761","limbic system development",0.198,1.7387,0.616,0.639,"specification of organ identity"),
c("GO:0072576","liver morphogenesis",0.024,1.4235,0.655,0.642,"specification of organ identity"),
c("GO:0042551","neuron maturation",0.076,1.5800,0.640,0.643,"specification of organ identity"),
c("GO:0042476","odontogenesis",0.418,1.7075,0.600,0.643,"specification of organ identity"),
c("GO:0055010","ventricular cardiac muscle tissue morphogenesis",0.089,2.4750,0.605,0.645,"specification of organ identity"),
c("GO:0048646","anatomical structure formation involved in morphogenesis",2.581,2.2218,0.646,0.651,"specification of organ identity"),
c("GO:0070268","cornification",0.005,1.8952,0.680,0.651,"specification of organ identity"),
c("GO:0048513","organ development",7.380,3.7447,0.579,0.662,"specification of organ identity"),
c("GO:0042475","odontogenesis of dentin-containing tooth",0.342,1.4460,0.605,0.664,"specification of organ identity"),
c("GO:0021846","cell proliferation in forebrain",0.079,1.6670,0.619,0.664,"specification of organ identity"),
c("GO:0048644","muscle organ morphogenesis",0.139,1.9108,0.626,0.673,"specification of organ identity"),
c("GO:0043585","nose morphogenesis",0.007,1.4235,0.675,0.679,"specification of organ identity"),
c("GO:0021869","forebrain ventricular zone progenitor cell division",0.013,1.5969,0.637,0.682,"specification of organ identity"),
c("GO:0050953","sensory perception of light stimulus",0.293,1.6388,0.746,0.682,"specification of organ identity"),
c("GO:0003228","atrial cardiac muscle tissue development",0.015,1.4235,0.683,0.683,"specification of organ identity"),
c("GO:0042492","gamma-delta T cell differentiation",0.039,1.3014,0.657,0.687,"specification of organ identity"),
c("GO:2000027","regulation of organ morphogenesis",0.521,1.4142,0.563,0.690,"specification of organ identity"),
c("GO:0030154","cell differentiation",8.015,2.0400,0.610,0.693,"specification of organ identity"),
c("GO:0032990","cell part morphogenesis",1.910,1.3271,0.592,0.695,"specification of organ identity"),
c("GO:0044707","single-multicellular organism process",15.092,2.2062,0.667,0.695,"specification of organ identity"),
c("GO:0010876","lipid localization",0.809,2.1073,0.958,0.000,"lipid localization"),
c("GO:0046717","acid secretion",0.178,1.8210,0.939,0.242,"lipid localization"),
c("GO:0034436","glycoprotein transport",0.003,1.4235,0.949,0.359,"lipid localization"),
c("GO:0071896","protein localization to adherens junction",0.008,1.4235,0.967,0.381,"lipid localization"),
c("GO:1902414","protein localization to cell junction",0.018,1.4235,0.965,0.404,"lipid localization"),
c("GO:0071205","protein localization to juxtaparanode region of axon",0.032,1.4235,0.960,0.422,"lipid localization"),
c("GO:0031630","regulation of synaptic vesicle fusion to presynaptic membrane",0.006,1.5969,0.824,0.458,"lipid localization"),
c("GO:0033326","cerebrospinal fluid secretion",0.014,1.5969,0.760,0.484,"lipid localization"),
c("GO:0014047","glutamate secretion",0.067,1.7652,0.902,0.541,"lipid localization"),
c("GO:0051877","pigment granule aggregation in cell center",0.003,1.4235,0.919,0.547,"lipid localization"),
c("GO:0060100","positive regulation of phagocytosis, engulfment",0.030,1.4235,0.834,0.556,"lipid localization"),
c("GO:0006869","lipid transport",0.732,1.8623,0.928,0.557,"lipid localization"),
c("GO:0016188","synaptic vesicle maturation",0.021,1.3014,0.704,0.666,"lipid localization"),
c("GO:0032501","multicellular organismal process",15.657,1.9154,0.996,0.000,"multicellular organismal process"),
c("GO:0032502","developmental process",13.055,1.7768,0.996,0.000,"developmental process"),
c("GO:0035634","response to stilbenoid",0.024,3.3188,0.941,0.000,"response to stilbenoid"),
c("GO:0035865","cellular response to potassium ion",0.007,1.5969,0.909,0.249,"response to stilbenoid"),
c("GO:0036006","cellular response to macrophage colony-stimulating factor stimulus",0.013,1.3014,0.900,0.303,"response to stilbenoid"),
c("GO:0080184","response to phenylpropanoid",0.042,2.8041,0.939,0.457,"response to stilbenoid"),
c("GO:0036005","response to macrophage colony-stimulating factor",0.013,1.3014,0.941,0.460,"response to stilbenoid"),
c("GO:0033210","leptin-mediated signaling pathway",0.015,1.3014,0.844,0.488,"response to stilbenoid"),
c("GO:2000121","regulation of removal of superoxide radicals",0.027,1.4235,0.813,0.512,"response to stilbenoid"),
c("GO:2000781","positive regulation of double-strand break repair",0.005,1.3014,0.834,0.532,"response to stilbenoid"),
c("GO:0035864","response to potassium ion",0.013,1.4235,0.946,0.541,"response to stilbenoid"),
c("GO:0040009","regulation of growth rate",0.005,1.5969,0.945,0.000,"regulation of growth rate"),
c("GO:0044130","negative regulation of growth of symbiont in host",0.079,1.3014,0.868,0.503,"regulation of growth rate"),
c("GO:0036149","phosphatidylinositol acyl-chain remodeling",0.018,2.3686,0.868,0.012,"phosphatidylinositol acyl-chain remodeling"),
c("GO:0052746","inositol phosphorylation",0.001,1.8952,0.904,0.145,"phosphatidylinositol acyl-chain remodeling"),
c("GO:0006629","lipid metabolic process",3.159,1.5173,0.921,0.162,"phosphatidylinositol acyl-chain remodeling"),
c("GO:0044240","multicellular organismal lipid catabolic process",0.003,1.8952,0.750,0.354,"phosphatidylinositol acyl-chain remodeling"),
c("GO:0090209","negative regulation of triglyceride metabolic process",0.009,1.5969,0.810,0.533,"phosphatidylinositol acyl-chain remodeling"),
c("GO:0006066","alcohol metabolic process",0.738,1.4508,0.907,0.546,"phosphatidylinositol acyl-chain remodeling"),
c("GO:0046486","glycerolipid metabolic process",0.810,2.0778,0.854,0.557,"phosphatidylinositol acyl-chain remodeling"),
c("GO:0036150","phosphatidylserine acyl-chain remodeling",0.019,2.1694,0.855,0.586,"phosphatidylinositol acyl-chain remodeling"),
c("GO:0006663","platelet activating factor biosynthetic process",0.005,1.4235,0.856,0.596,"phosphatidylinositol acyl-chain remodeling"),
c("GO:0044241","lipid digestion",0.028,1.3014,0.783,0.655,"phosphatidylinositol acyl-chain remodeling"),
c("GO:0046469","platelet activating factor metabolic process",0.006,1.3014,0.862,0.672,"phosphatidylinositol acyl-chain remodeling"),
c("GO:0008283","cell proliferation",4.075,1.7467,0.966,0.019,"cell proliferation"),
c("GO:0090074","negative regulation of protein homodimerization activity",0.005,1.4235,0.965,0.033,"negative regulation of protein homodimerization activity"),
c("GO:0032237","activation of store-operated calcium channel activity",0.013,1.3014,0.871,0.275,"negative regulation of protein homodimerization activity"),
c("GO:0032780","negative regulation of ATPase activity",0.010,1.3014,0.847,0.404,"negative regulation of protein homodimerization activity"),
c("GO:0097006","regulation of plasma lipoprotein particle levels",0.120,1.3098,0.963,0.042,"regulation of plasma lipoprotein particle levels"),
c("GO:0000320","re-entry into mitotic cell cycle",0.007,1.4235,0.939,0.048,"re-entry into mitotic cell cycle"),
c("GO:0032107","regulation of response to nutrient levels",0.109,2.4750,0.901,0.052,"regulation of response to nutrient levels"),
c("GO:0032104","regulation of response to extracellular stimulus",0.109,2.4750,0.909,0.652,"regulation of response to nutrient levels"),
c("GO:0050728","negative regulation of inflammatory response",0.236,1.6386,0.843,0.672,"regulation of response to nutrient levels"),
c("GO:0031146","SCF-dependent proteasomal ubiquitin-dependent protein catabolic process",0.027,2.2628,0.924,0.060,"SCF-dependent proteasomal ubiquitin-dependent protein catabolism"),
c("GO:0018350","protein esterification",0.001,1.8952,0.942,0.181,"SCF-dependent proteasomal ubiquitin-dependent protein catabolism"),
c("GO:0030166","proteoglycan biosynthetic process",0.088,1.9427,0.908,0.193,"SCF-dependent proteasomal ubiquitin-dependent protein catabolism"),
c("GO:0046491","L-methylmalonyl-CoA metabolic process",0.001,1.8952,0.904,0.218,"SCF-dependent proteasomal ubiquitin-dependent protein catabolism"),
c("GO:0018342","protein prenylation",0.041,1.3014,0.926,0.222,"SCF-dependent proteasomal ubiquitin-dependent protein catabolism"),
c("GO:0052646","alditol phosphate metabolic process",0.079,1.5024,0.923,0.283,"SCF-dependent proteasomal ubiquitin-dependent protein catabolism"),
c("GO:0030210","heparin biosynthetic process",0.009,1.3014,0.883,0.471,"SCF-dependent proteasomal ubiquitin-dependent protein catabolism"),
c("GO:0030202","heparin metabolic process",0.009,1.3014,0.896,0.624,"SCF-dependent proteasomal ubiquitin-dependent protein catabolism"),
c("GO:0006029","proteoglycan metabolic process",0.173,1.3898,0.916,0.693,"SCF-dependent proteasomal ubiquitin-dependent protein catabolism"),
c("GO:0097354","prenylation",0.041,1.3014,0.954,0.063,"prenylation"),
c("GO:0016055","Wnt signaling pathway",0.887,2.3979,0.836,0.069,"Wnt signaling pathway"),
c("GO:0030030","cell projection organization",2.744,2.0721,0.883,0.104,"Wnt signaling pathway"),
c("GO:2000639","negative regulation of SREBP signaling pathway",0.001,1.8952,0.816,0.191,"Wnt signaling pathway"),
c("GO:0042127","regulation of cell proliferation",3.148,2.0921,0.838,0.215,"Wnt signaling pathway"),
c("GO:0034721","histone H3-K4 demethylation, trimethyl-H3-K4-specific",0.001,1.8952,0.892,0.244,"Wnt signaling pathway"),
c("GO:0007296","vitellogenesis",0.003,1.5969,0.758,0.264,"Wnt signaling pathway"),
c("GO:0048790","maintenance of presynaptic active zone structure",0.003,1.8952,0.925,0.264,"Wnt signaling pathway"),
c("GO:0035602","fibroblast growth factor receptor signaling pathway involved in negative regulation of apoptotic process in bone marrow",0.001,1.8952,0.826,0.273,"Wnt signaling pathway"),
c("GO:0048015","phosphatidylinositol-mediated signaling",0.367,1.3723,0.846,0.297,"Wnt signaling pathway"),
c("GO:0038026","reelin-mediated signaling pathway",0.005,1.5969,0.882,0.299,"Wnt signaling pathway"),
c("GO:2000210","positive regulation of anoikis",0.005,1.4235,0.882,0.313,"Wnt signaling pathway"),
c("GO:0043954","cellular component maintenance",0.032,1.8781,0.915,0.314,"Wnt signaling pathway"),
c("GO:0050731","positive regulation of peptidyl-tyrosine phosphorylation",0.330,1.7994,0.816,0.335,"Wnt signaling pathway"),
c("GO:0019934","cGMP-mediated signaling",0.014,1.3014,0.876,0.348,"Wnt signaling pathway"),
c("GO:0051703","intraspecies interaction between organisms",0.109,1.5800,0.970,0.348,"Wnt signaling pathway"),
c("GO:0055096","low-density lipoprotein particle mediated signaling",0.017,1.5969,0.844,0.355,"Wnt signaling pathway"),
c("GO:0034141","positive regulation of toll-like receptor 3 signaling pathway",0.005,1.5969,0.842,0.361,"Wnt signaling pathway"),
c("GO:0016310","phosphorylation",10.160,1.3801,0.906,0.362,"Wnt signaling pathway"),
c("GO:1990134","epithelial cell apoptotic process involved in palatal shelf morphogenesis",0.001,1.8952,0.741,0.390,"Wnt signaling pathway"),
c("GO:0045785","positive regulation of cell adhesion",0.418,1.3363,0.882,0.402,"Wnt signaling pathway"),
c("GO:0070339","response to bacterial lipopeptide",0.056,1.4235,0.902,0.416,"Wnt signaling pathway"),
c("GO:0016567","protein ubiquitination",1.323,1.5694,0.906,0.417,"Wnt signaling pathway"),
c("GO:0071839","apoptotic process in bone marrow",0.008,1.4235,0.928,0.426,"Wnt signaling pathway"),
c("GO:0007166","cell surface receptor signaling pathway",8.919,1.3724,0.810,0.430,"Wnt signaling pathway"),
c("GO:0030111","regulation of Wnt signaling pathway",0.482,1.8881,0.819,0.438,"Wnt signaling pathway"),
c("GO:0048017","inositol lipid-mediated signaling",0.367,1.3723,0.846,0.450,"Wnt signaling pathway"),
c("GO:0071838","cell proliferation in bone marrow",0.008,1.5969,0.943,0.476,"Wnt signaling pathway"),
c("GO:0035603","fibroblast growth factor receptor signaling pathway involved in hemopoiesis",0.001,1.8952,0.663,0.492,"Wnt signaling pathway"),
c("GO:0035604","fibroblast growth factor receptor signaling pathway involved in positive regulation of cell proliferation in bone marrow",0.001,1.8952,0.820,0.492,"Wnt signaling pathway"),
c("GO:0035607","fibroblast growth factor receptor signaling pathway involved in orbitofrontal cortex development",0.006,1.5969,0.603,0.529,"Wnt signaling pathway"),
c("GO:1902533","positive regulation of intracellular signal transduction",1.490,1.4557,0.788,0.559,"Wnt signaling pathway"),
c("GO:0034720","histone H3-K4 demethylation",0.019,1.3014,0.875,0.563,"Wnt signaling pathway"),
c("GO:0038018","Wnt receptor catabolic process",0.002,1.5969,0.824,0.585,"Wnt signaling pathway"),
c("GO:0045944","positive regulation of transcription from RNA polymerase II promoter",2.089,1.5287,0.815,0.598,"Wnt signaling pathway"),
c("GO:0070391","response to lipoteichoic acid",0.027,1.3014,0.908,0.621,"Wnt signaling pathway"),
c("GO:0071223","cellular response to lipoteichoic acid",0.027,1.3014,0.869,0.621,"Wnt signaling pathway"),
c("GO:0021769","orbitofrontal cortex development",0.008,1.4235,0.682,0.647,"Wnt signaling pathway"),
c("GO:0032493","response to bacterial lipoprotein",0.058,1.4235,0.905,0.653,"Wnt signaling pathway"),
c("GO:0003341","cilium movement",0.085,1.7652,0.906,0.073,"cilium movement"),
c("GO:0010826","negative regulation of centrosome duplication",0.015,1.5969,0.826,0.494,"cilium movement"),
c("GO:0032876","negative regulation of DNA endoreduplication",0.006,1.3014,0.836,0.560,"cilium movement"),
c("GO:0003352","regulation of cilium movement",0.015,1.4235,0.858,0.654,"cilium movement"));

stuff <- data.frame(revigo.data);
names(stuff) <- revigo.names;

stuff$abslog10pvalue <- as.numeric( as.character(stuff$abslog10pvalue) );
stuff$freqInDbPercent <- as.numeric( as.character(stuff$freqInDbPercent) );
stuff$uniqueness <- as.numeric( as.character(stuff$uniqueness) );
stuff$dispensability <- as.numeric( as.character(stuff$dispensability) );

# by default, outputs to a PDF file
pdf( file="revigo_treemap.pdf", width=16, height=9 ) # width and height are in inches

# check the tmPlot command documentation for all possible parameters - there are a lot more
tmPlot(
	stuff,
	index = c("representative","description"),
	vSize = "abslog10pvalue",
	type = "categorical",
	vColor = "representative",
	title = "REVIGO Gene Ontology treemap",
	inflate.labels = FALSE,      # set this to TRUE for space-filling group labels - good for posters
	lowerbound.cex.labels = 0,   # try to draw as many labels as possible (still, some small squares may not get a label)
	bg.labels = "#CCCCCCAA",     # define background color of group labels
												       # "#CCCCCC00" is fully transparent, "#CCCCCCAA" is semi-transparent grey, NA is opaque
	position.legend = "none"
)

dev.off()
